torch/csrc/jit/interpreter.cpp

```diff
@@ -19,6 +19,270 @@ namespace py = pybind11;
+    for(auto n : b->nodes()) {
+      if(n->kind() == kLoop) {
+
+        // remove the trip count from Loop inputs, we don't support it yet
+        n->removeInput(0);
+        JIT_ASSERT(n->blocks()[0]->inputs()[0]->uses().size() == 0 &&
+          "NYI - use of trip count variable");
+        JIT_ASSERT(n->blocks()[0]->inputs()[1]->uses().size() == 0 &&
+          "NYI - use of cond variable in loop");
+
+        // TODO: remove cond as input to loop carries, it just complicates this
+        // implementation
+        n->blocks()[0]->eraseInput(1);
+        n->blocks()[0]->eraseInput(0);
+      }
+      for(auto sb : n->blocks()) {
+        desugarTripCounts(sb);
+      }
+    }
+  }
+  Node * prependLocation() {
+    //handle corner case where there are no nodes,
+    if(graph->nodes().begin() == graph->nodes().end()) {
+      return graph->return_node();
+    }
+    return *graph->nodes().begin();
+  }
+  // removes all inputs and outputs to a graph, replacing them with nodes before of after each insertStage
+  void flattenStages() {
+    WithInsertPoint guard(*graph, prependLocation());
+    size_t input_pos = 0;
+    size_t output_pos = 0;
+    auto it = graph->nodes().begin();
+    for(size_t i = 0; i <= graph->stage(); i++) {
+      stage_input_types.emplace_back();
+      auto store = graph->insertNode(graph->create(kStore, 0));
+      // TODO: unused inputs need drop nodes added for them.
+      while(input_pos < graph->inputs().size() && graph->inputs()[input_pos]->stage() == i) {
+        auto nv = store->addOutput();
+        auto old_node = graph->inputs()[input_pos];
+        stage_input_types[i].push_back(old_node->typeOption());
+        old_node->replaceAllUsesWith(nv);
+        input_pos++;
+      }
+      while(it != graph->nodes().end() && it->stage() == i)
+        ++it;
+      graph->setInsertPoint(*it);
+      auto load = graph->insertNode(graph->create(kLoad, 0));
+      while(output_pos < graph->outputs().size() && graph->outputs()[output_pos]->stage() == i) {
+        load->addInput(graph->outputs()[output_pos]);
+        output_pos++;
+      }
+    }
+    while (graph->inputs().size() > 0)
+      graph->eraseInput(graph->inputs().size() - 1);
+    while (graph->outputs().size() > 0)
+      graph->eraseOutput(graph->outputs().size() - 1);
+  }
+
+  Node * findOrCreateDropInstructionForNode(Node * n) {
+    auto it = drop_for_node.find(n);
+    if(it == drop_for_node.end()) {
+      auto drop_node = graph->create(kDrop, 0);
+      drop_node->insertAfter(n);
+      it = drop_for_node.emplace(n, drop_node).first;
+    }
+    return it->second;
+  }
+
+  // finds the node in the _same block_ as 'definition' that contains
+  // the 'n' Node
+  // prereq: one of n's inputs must be 'definition' to ensure it is in scope
+  // e.g.
+  // n0: a = 4
+  // n1: if <cond>:
+  // n2:    b = a + a
+  // findNodeInTheSameBlockAs(n2, a) == n1
+  Node * findNodeInTheSameBlockAs(Node * n, Value * definition) {
```

Comment:
Since the only way you're using the `definition` is to get it's `owningBlock()`, I think it would make the semantics clearer if this method was `findOwnerInBlock(Node *n, Block *block)` (nice to explicitly say that it's a parent/owner of `n`, not just a node)